### DISCUSSION TITLE
[PluginDev] ForeignKey from Core to Plugin

### DISCUSSION THREAD

**Discussion by AlexandreLanglade** [netbox-community/netbox#9219] [Q&A] [answered]
Hi everyone !
It's possible to dev a custom model "A" in a plugin that has a field (foreign key) pointing the model dcim.Device. But I want the opposite.
My question is : Is there a way to make the dcim.Device model pointing at the "A" model **without modifying Netbox's source code** ?

Ty :smiley:

**Comment by jeremystretch** [comment, marked as answer]
You can add a [ManyToManyField](https://docs.djangoproject.com/en/4.0/ref/models/fields/#manytomanyfield) or [OneToOneField](https://docs.djangoproject.com/en/4.0/ref/models/fields/#onetoonefield) to your model to represent the inverse relationship. You might also consider employing a [custom object field](https://docs.netbox.dev/en/stable/customization/custom-fields/#custom-object-fields), depending on the specific use case.
